sql/logictest: TestLogic/fakedist-spec-planning/numeric_references failed

sql/logictest.TestLogic/fakedist-spec-planning/numeric_references [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3594232&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3594232&tab=artifacts#/) on master @ [b3af96b0686773c78325d2b8b0623a8fcd3e9bf2](https://github.com/cockroachdb/cockroach/commits/b3af96b0686773c78325d2b8b0623a8fcd3e9bf2):


```
Slow failing tests:
TestLogic/fakedist-spec-planning/numeric_references - 0.00s

Slow passing tests:
<none>
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestLogic/fakedist-spec-planning/numeric_references PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=2
</p>
</details>
/cc @cockroachdb/sql-queries jordanlewis
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic/fakedist-spec-planning/numeric_references.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/logictest.TestLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3596602&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3596602&tab=artifacts#/) on master @ [5d972a683c531326bc0af403c6d3373d3f4b2267](https://github.com/cockroachdb/cockroach/commits/5d972a683c531326bc0af403c6d3373d3f4b2267):


```
=== RUN   TestLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic962287550
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestLogic
    logic.go:3437: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic962287550
--- FAIL: TestLogic (2547.27s)
=== RUN   TestLogic/5node/experimental_distsql_planning_5node
    logic.go:2574: 
        
        testdata/logic_test/experimental_distsql_planning_5node:14: 
        expected success, but found
        (XXUUU) cannot up-replicate to s3; missing gossiped StoreDescriptor (the store is likely dead, draining or decommissioning)
--- done: testdata/logic_test/experimental_distsql_planning_5node with config 5node: 1 tests, 1 failures
    logic.go:3205: 
        testdata/logic_test/experimental_distsql_planning_5node:14: error while processing
    logic.go:3205: pq: cannot up-replicate to s3; missing gossiped StoreDescriptor (the store is likely dead, draining or decommissioning)
        --- FAIL: TestLogic/5node/experimental_distsql_planning_5node (16.17s)
=== RUN   TestLogic/5node
    --- FAIL: TestLogic/5node (52.36s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=2
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #63262 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-20.2]
- #61657 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-21.1]
- #58269 sql/logictest: TestLogic failed [C-test-failure O-robot T-sql-schema branch-release-20.1]
</p>
</details>
/cc @cockroachdb/sql-queries jordanlewis
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/logic_test.go
+++ b/pkg/sql/logictest/logic_test.go
@@ -13,6 +13,7 @@ package logictest
 import (
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 )
 
@@ -26,6 +27,7 @@ import (
 // See the comments in logic.go for more details.
 func TestLogic(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	skip.UnderDeadlock(t, "times out and/or hangs")
 	RunLogicTest(t, TestServerArgs{}, "testdata/logic_test/[^.]*")
 }
 

--- a/pkg/sql/opt/exec/execbuilder/BUILD.bazel
+++ b/pkg/sql/opt/exec/execbuilder/BUILD.bazel
@@ -67,6 +67,7 @@ go_test(
         "//pkg/sql/opt",
         "//pkg/sql/sem/tree",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/testcluster",
         "//pkg/util/leaktest",
         "//pkg/util/randutil",

--- a/pkg/sql/opt/exec/execbuilder/builder_test.go
+++ b/pkg/sql/opt/exec/execbuilder/builder_test.go
@@ -15,6 +15,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/logictest"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 )
 
@@ -27,5 +28,6 @@ import (
 func TestExecBuild(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer sql.TestingOverrideExplainEnvVersion("CockroachDB execbuilder test version")()
+	skip.UnderDeadlock(t, "times out and/or hangs")
 	logictest.RunLogicTest(t, logictest.TestServerArgs{}, "testdata/[^.]*")
 }

